packages/help-center/src/hooks/use-container-click.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useMemo } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+const isThisASupportArticleLink = ( href: string ) =>
+	/wordpress\.com(\/\w\w)?(?=\/support\/)|support\.wordpress\.com/.test( href );
+
+export const useContainerClick = ( node: HTMLDivElement | null ) => {
+	const navigate = useNavigate();
+
+	const filters = useMemo(
+		() => [
+			/**
+			 * Make support article links open within Help Center.
+			 */
+			{
+				condition: ( target: HTMLElement ) =>
+					target.tagName === 'A' &&
+					isThisASupportArticleLink( target.getAttribute( 'href' ) || '' ),
+				action: ( event: MouseEvent, target: HTMLElement ) => {
+					event.preventDefault();
+					const href = target.getAttribute( 'href' );
+					navigate( `/post?link=${ href }` );
+				},
+			},
+
+			/**
+			 * Fix table of content links.
+			 */
+			{
+				condition: ( target: HTMLElement ) =>
+					target.tagName === 'A' && target.closest( '.a8c-table-of-contents' ),
+				action: ( event: MouseEvent, target: HTMLElement ) => {
+					event.preventDefault();
+					const href = target.getAttribute( 'href' );
+					const hash = href?.split( '#' )[ 1 ];
+					node?.querySelector( `#${ hash }` )?.scrollIntoView();
```

Comment:
You can use `event.currentTarget` instead of accessing `node` that can be outdated.